server/lib/topic_cache.py

```diff
@@ -154,7 +191,32 @@ def load_files(fpath_list: List[str], name_overrides: Dict,
 
 
 def load(name_overrides: Dict) -> Dict[str, TopicCache]:
-  topic_cache_map = {}
+  topic_cache_map: dict[str, TopicCache] = {}
   for dc, fpath_list in TOPIC_CACHE_FILES.items():
     topic_cache_map[dc] = load_files(fpath_list, name_overrides, dc)
+
+  if is_custom_dc():
+    custom_dc_topic_cache = _load_custom_dc_topic_cache(name_overrides)
+    if custom_dc_topic_cache:
+      topic_cache_map[DCNames.CUSTOM_DC.value] = custom_dc_topic_cache
+      topic_cache_map[DCNames.MAIN_DC.value]._merge(custom_dc_topic_cache)
+
   return topic_cache_map
+
+
+def _load_custom_dc_topic_cache(name_overrides: dict) -> TopicCache:
+  local_path = _get_local_custom_dc_topic_cache_path()
+  if not local_path:
+    return None
+  return load_files([local_path], name_overrides, DCNames.CUSTOM_DC.value)
+
+
+def _get_local_custom_dc_topic_cache_path() -> str:
+  path = get_topic_cache_path()
+  if not path:
+    logging.info("No Custom DC topic cache path specified.")
+    return path
+  logging.info("Custom DC topic cache will be downloaded from: %s", path)
```

Comment:
nit: Should the "download" msg be under if is_gcs_path?

Reply:
I wanted to log in both cases. Changed "downloaded" to just "loaded".